lib/make-spawn-args.js

```diff
@@ -1,7 +1,16 @@
 /* eslint camelcase: "off" */
 const setPATH = require('./set-path.js')
 const { resolve } = require('path')
-const npm_config_node_gyp = require.resolve('node-gyp/bin/node-gyp.js')
+let npm_config_node_gyp
+try {
+  /* istanbul ignore next */
+  if (typeof require === 'function' && typeof require.resolve === 'function') {
```

Comment:
The `typeof require` may seem redundant here considering we call `require('path')` just prior.

Rightfully or not, bundlers like webpack and esbuild may transform the direct `require()` calls but leave `require.resolve` intact.